File: SOURCE_WEBAPI/VecompSoftware.DocSuite.Public.Core.Models/Domains/PECMails/PECMailModel.cs
using System;
using System.Collections.Generic;

namespace VecompSoftware.DocSuite.Public.Core.Models.Domains.PECMails
{
    public class PECMailModel : DomainModel
    {
        #region [ Contructors ]
        public PECMailModel(Guid id)
            : base(id)
        {
            Receipts = new HashSet<PECMailReceiptModel>();
        }
        #endregion

        #region [ Properties ]

        public short? Year { get; set; }

        public int? Number { get; set; }

        public string Subject { get; set; }

        public PECDirection Direction { get; set; }

        public string Senders { get; set; }

        public string Recipients { get; set; }

        public PECActiveType IsActive { get; set; }

        public DateTime? Date { get; set; }

        public string Body { get; set; }

        public string Step { get; set; }

        public PECType? PECType { get; set; }

        public ICollection<PECMailReceiptModel> Receipts { get; set; }

        #endregion
    }
}

File: SOURCE_EXTERNAL/VecompSoftware.DocSuite.ExternalViewer/src/app/models/pec-mails/pec-mail.model.ts
import { IdentifierModel } from '../identifier.model';
import { Environment } from '../environment';
import { ProtocolModel } from '../protocols/protocol.model';
import { PECType } from './pec.type';
import { PECActiveType } from './pec-active.type';
import { PECDirectionType } from './pec-direction.type';
import { PECMailReceiptModel } from './pec-mail-receipt.model';
import { BaseServiceModel } from '../base-service.model';

export class PECMailModel implements IdentifierModel, BaseServiceModel {
    constructor() {
        this.serviceModelName = "PECMailModel";
    }

    id: string;
    year?: number;
    number?: number;
    direction: PECDirectionType;
    subject: string;
    senders: string;
    recipients: string;
    active: PECActiveType;
    date?: Date;
    type: PECType;
    body: string;
    step: string;
    readonly serviceModelName: string;

    receipts: PECMailReceiptModel[];

}